Tests/StreamChatUITests/AssertSnapshot.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import SnapshotTesting
+import XCTest
+
+/// Snapshot of a view controller. All variants will be tested by default. For each variant, it will take a snapshot.
+/// It is recommended to test all trait variants only for view controllers, and for small view components only test
+/// the default variant. This is to minimize the amount of pictures stored in the repo.
+///
+/// - Parameters:
+///   - vc: The view controller to be tested.
+///   - isEmbededInNavigationController: If the view controller should be embeded in a navigation controller.
+///   Useful to test the navigation bar.
+///   - variants: The variants that a snapshot will be taken. All variants should be tested for a view controller.
+///   - record: True if a new reference should be saved. False by default,
+///   so that the newly captured snapshot is compared with the current reference.
+func AssertSnapshot(
+    _ vc: UIViewController,
+    isEmbededInNavigationController: Bool = false,
+    variants: [SnapshotVariation] = SnapshotVariation.allCases,
+    record: Bool = false,
+    line: UInt = #line,
+    file: StaticString = #file,
+    function: String = #function
+) {
+    let viewController = isEmbededInNavigationController ? UINavigationController(rootViewController: vc) : vc
+    variants.forEach { variant in
+        assertSnapshot(
+            matching: viewController,
+            as: .image(traits: variant.trait),
+            named: variant.rawValue,
+            record: record,
+            file: file,
+            testName: function,
+            line: line
+        )
+    }
+}
+
+/// Snapshot of a view. Used for view components. Only the default variant is tested.
+/// To minimize the amount of snapshot pictures saved in the repo, the rest of the variants should
+/// be tested in view controllers.
+///
+/// - Parameters:
+///   - view: The component to be tested.
+///   - variants: The variants that a snapshot will be taken.
+///   Only the light and dark variants should be tested in small components.
+///   - record: True if a new reference should be saved. False by default,
+///   so that the newly captured snapshot is compared with the current reference.
+func AssertSnapshot(
+    _ view: UIView,
+    variants: [SnapshotVariation] = [.light, .dark],
+    record: Bool = false,
+    line: UInt = #line,
+    file: StaticString = #file,
+    function: String = #function
+) {
+    variants.forEach { variant in
+        assertSnapshot(
+            matching: view,
+            as: .image(traits: variant.trait),
+            named: variant.rawValue,
+            record: record,
+            file: file,
+            testName: function,
+            line: line
+        )
+    }
+}
```

Comment:
If there's a view component that has too many logic, then, we could test it in isolation. For example, a message cell type, probably it will not be a good idea to create the MessageListVC with all types of messages, because they will not even fit in the screen.